src/components/lunchRequirement.jsx

```diff
@@ -4,11 +4,16 @@ import Typography from "@mui/material/Typography";
 import Divider from "@mui/material/Divider";
 import TextField from "@mui/material/TextField";
 import ButtonGroup from "./buttonGroup";
+import { useSelector } from "react-redux";
 export default function LunchRequirement() {
-  const [userName, setUserName] = useState("");
   const [itemDescription, setItemDescription] = useState("");
   const [roti, setRoti] = useState("");
   const [amount, setAmount] = useState("");
+  const user = useSelector((state) => {
```

Comment:
You can simplify, const user = useSelector((state) => state?.signIn?.signIn?.userName ) || ""